tfe/data_source_github_app_installation.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceTFEGHAInstallation() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGHAInstallationRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
```

Comment:
At first look over the installation API request made, it seems like the `id` and `installation_id` is pointing to the same value, what is the difference between these two values?

Also, the current pattern in the provider is to let terraform state own the `id` field, in other existing resources, the user doesn't set the `id` since it is always computed as opposed to user provided. Suggestion is to consider not allowing user to set the `id` attribute.

Reply:
` id `- is the github app installation external id (formatted like "ghain-xxxx" , thats generated by TFC when the GHA installation is created for the first time using the UI.
`installation_id` - is the github's reference for the same installation which is a numeric field.
The purpose of this data block is to fetch the `id` using either installation `name` or `installation_id`.